pkg/vm/vm_test.go

```diff
@@ -3331,6 +3331,110 @@ func TestBitAndNumericOpcodes(t *testing.T) {
 	}
 }
 
+var stackIsolationTestCases = map[opcode.Opcode][]struct {
+	name   string
+	params []interface{}
+}{
+	opcode.CALLE: {
+		{
+			name: "no_params",
+		},
+		{
+			name:   "with_params",
+			params: []interface{}{big.NewInt(5), true, []byte{1, 2, 3}},
+		},
+	},
+	opcode.CALLED: {
+		{
+			name: "no_paranms",
+		},
+		{
+			name:   "with_params",
+			params: []interface{}{big.NewInt(5), true, []byte{1, 2, 3}},
+		},
+	},
+	opcode.CALLET: {
+		{
+			name: "no_params",
+		},
+		{
+			name:   "with_params",
+			params: []interface{}{big.NewInt(5), true, []byte{1, 2, 3}},
+		},
+	},
+	opcode.CALLEDT: {
+		{
+			name: "no_params",
+		},
+		{
+			name:   "with_params",
+			params: []interface{}{big.NewInt(5), true, []byte{1, 2, 3}},
+		},
+	},
+}
+
+func TestStackIsolationOpcodes(t *testing.T) {
+	scriptHash := util.Uint160{1, 2, 3}
+	for code, codeTestCases := range stackIsolationTestCases {
+		t.Run(code.String(), func(t *testing.T) {
+			for _, testCase := range codeTestCases {
+				t.Run(testCase.name, func(t *testing.T) {
+					rvcount := len(testCase.params) + 1 // parameters + DEPTH
+					pcount := len(testCase.params)
+					prog := []byte{byte(code), byte(rvcount), byte(pcount)}
+					if code == opcode.CALLE || code == opcode.CALLET {
+						prog = append(prog, scriptHash.BytesBE()...)
+					}
+					prog = append(prog, byte(opcode.RET))
+
+					vm := load(prog)
+					vm.SetScriptGetter(func(in util.Uint160) ([]byte, bool) {
+						if in.Equals(scriptHash) {
+							return makeProgram(opcode.DEPTH), true
+						}
+						return nil, false
+					})
+					if code == opcode.CALLED || code == opcode.CALLEDT {
+						vm.Context().hasDynamicInvoke = true
+					}
+					if code == opcode.CALLET || code == opcode.CALLEDT {
+						vm.Context().rvcount = rvcount
+					}
+					// add extra parameter just to check that it won't appear on new estack
+					vm.estack.PushVal(7)
+					for _, param := range testCase.params {
+						vm.estack.PushVal(param)
+					}
+					if code == opcode.CALLED || code == opcode.CALLEDT {
+						vm.estack.PushVal(scriptHash.BytesBE())
+					}
+
+					runVM(t, vm)
+					elem := vm.estack.Pop() // depth should be equal to params count, as it's a separate execution context
+					assert.Equal(t, int64(pcount), elem.BigInt().Int64())
+					for i := pcount; i > 0; i-- {
+						p := vm.estack.Pop()
+						assert.Equal(t, testCase.params[i-1], p.value.Value())
+					}
+				})
+			}
+		})
+	}
+}
+
+func TestCALLI(t *testing.T) {
+	prog := []byte{byte(opcode.PUSHBYTES2), byte(opcode.Opcode(uint16(1))), byte(opcode.Opcode(uint16(1) >> 8)), // little-endian
```

Comment:
`byte(opcode.Opcode(uint16(1)))` is just a `1`, really. And it makes no sense doing `uint16(1) >> 8` as it's just a `0`.